### DISCUSSION TITLE
New tutorials on the surface finite element method

### DISCUSSION THREAD

**Discussion by tranner** [firedrakeproject/firedrake#2553] [Show and tell]
Hi everyone,

I wanted to share with you some tutorial notebooks which implement some nice test cases for the surface finite element method. These notebooks guide you through implementations of the surface finite element method for an elliptic equation on a curved surface, an advection diffusion equation on an evolving surface and mean curvature flow of an evolving parametrised surface.

You can try out the notebooks through Binder:
[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gl/tom-ranner%2Fsfem-tutorial/main)

Otherwise the source files can be found on gitlab:
https://gitlab.com/tom-ranner/sfem-tutorial

Please let me know if you would prefer me to post this message somewhere else.

The notebooks formed part of some tutorials I recently delivered on the surface finite element at the [surfaces22 summer school](https://www.tu-chemnitz.de/mathematik/harmonische_analysis/events/surfaces22/).

Thanks,
Tom

**Comment by francesco-ballarin** [comment]
Very cool!

Apart from the fact that I like the topic itself, from a technical standpoint I especially like the integration with pyvista. 

FYI, a bit of unsolicited advertisement: if some of your users want to use Colab instead of binder you may want to have a look at a project of mine, [FEM on Colab](https://fem-on-colab.github.io/), that a few firedrake developers have already used in the past to deliver tutorials during schools.
